pkg/workers/workers.go

```diff
@@ -22,32 +20,29 @@ type ScratchDataWorker struct {
 	destinationManager *destinations.DestinationManager
 }
 
-func (w *ScratchDataWorker) Start(ctx context.Context, threadId int) {
+func (w *ScratchDataWorker) Start(ctx context.Context, threadId int, msgs <-chan []byte) {
 	log.Debug().Int("thread", threadId).Msg("Starting worker")
 
 	for {
-		item, ok := w.StorageServices.Queue.Dequeue()
-
-		if !ok {
-			time.Sleep(1 * time.Second)
-		} else {
-			message, err := w.messageToStruct(item)
+		select {
+		case <-ctx.Done():
```

Comment:
I actually intentionally don't use a channel to dequeue messages, and here is why. (tell me if there is a logic error.) Suppose this sequence of events:

1. queue.Dequeue() fetches a message from SQS
2. We attempt to send that message to the channel, but it blocks because the worker is busy
3. We ctrl-c the program

In this case, I think we'd lose a message. ctx.Done() will trigger, this worker will return and the program will terminate. Meanwhile we've already fetched a message from SQS, but it hasn't been consumed by anyone. 

(No need to fix this now, just something to talk through!)

Reply:
yes, this is a failure scenario. this is actually something I noticed in the code. The message should only be deleted from the queue if it is handled successfully https://chat.openai.com/share/1b1b2c58-6ee5-4b3f-b0f3-67260eb71780

The queue delete needs to be moved out of this function and is called after a successful processing https://github.com/scratchdata/scratchdata/blob/main/pkg/storage/queue/sqs/sqs.go#L75

Reply:
That seems like a lot of work just so we can a channel, doesn't it? To me, the question is whether we want consumers from the queue to have to contend with a `Delete()` operation in addition to Dequeue?

Other things to think about: What timeout should we set before SQS automatically re-queues an unacknowledged message? If we end up using Redis as a queue in the future (ie, `BLPOP`) then consuming+deleting are the same action.